activities.el

```diff
@@ -479,12 +479,47 @@ It will not be recoverable."
   "Automatically save activities when Emacs has been idle this many seconds."
   :type 'natnum)
 
+(defvar activities-mode-map (make-sparse-keymap)
+  "The mode keymap for `activities-mode'.")
+
+(defun activities-mode-line-format ()
+  (when-let ((cur (activities-current)))
+    (let ((cur-activity-title (concat " " (activities-name-for cur))))
+      `(:propertize
+        ,cur-activity-title
+        mouse-face mode-line-highlight
+        help-echo
+        ,(lambda (&rest _)
+           (concat
+            (format "Current activity:%s\n" cur-activity-title)
+            "mouse-1: Display minor mode menu\n"
+            "mouse-2: Show help for minor mode"))
+        keymap
+        ,(let ((map (make-sparse-keymap)))
+           (define-key map [mode-line down-mouse-1]
+                       activities-mode-line-menu)
+           (define-key map [mode-line down-mouse-3]
+                       activities-mode-line-menu)
+           (define-key map [mode-line mouse-2]
+                       (lambda ()
+                         (interactive)
+                         (describe-function 'activities-mode)))
+           map)))))
+
+(defcustom activities-mode-line '(:eval (activities-mode-line-format))
+  "Activities mode line definition."
+  :type 'sexp
+  :group 'activities
+  :risky t)
+
 ;;;###autoload
 (define-minor-mode activities-mode
   "Automatically remember activities' state.
 accordingly."
   :global t
   :group 'activities
+  :lighter activities-mode-line
+  :keymap activities-mode-map
```

Comment:
See earlier comment about menu bar and mode map.

Alternatively, we could consider defining this map as a prefix map for the suggested bindings.

Reply:
Let's address prefix map and suggested bindings separately.